src/main.rs

```diff
@@ -402,117 +475,184 @@ fn get_args(matches: &ArgMatches) -> CliArgs {
-        MOUNT_POINT_ARG_NAME,
-        args.mount_dir,
-        RUN_AS_ARG_NAME,
-        args.run_as,
-        ETCD_ADDRESS_ARG_NAME,
-        args.etcd_address_vec,
-        SCHEDULER_EXTENDER_PORT_ARG_NAME,
-        args.scheduler_extender_port,
-        CACHE_CAPACITY_ARG_NAME,
-        args.cache_capacity,
-        SERVER_PORT_NUM_ARG_NAME,
-        args.server_port,
-        VOLUME_INFO_ARG_NAME,
-        args.volume_info,
-        VOLUME_TYPE_ARG_NAME,
-        args.volume_type,
-    );
 
-    let args_clone = args.clone();
-    let etcd_delegate = EtcdDelegate::new(args.etcd_address_vec)?;
-    let etcd_delegate_clone = etcd_delegate.clone();
-    let meta_data = csi::build_meta_data(
-        args.worker_port,
-        args.node_id,
-        args.ip_address,
-        args.mount_dir,
-        args.run_as,
-        etcd_delegate,
-    )?;
-    let md = Arc::new(meta_data);
-    let async_server = true;
-
-    match args.run_as {
-        RunAsRole::Controller => {
-            let controller_server = csi::build_grpc_controller_server(
-                &args.end_point,
-                &args.driver_name,
-                Arc::<MetaData>::clone(&md),
-            )?;
-            csi::run_grpc_servers(&mut [controller_server], async_server);
-        }
-        RunAsRole::Node => {
-            let worker_server = csi::build_grpc_worker_server(Arc::<MetaData>::clone(&md))?;
-            let node_server = csi::build_grpc_node_server(&args.end_point, &args.driver_name, md)?;
-            let csi_thread = std::thread::spawn(move || {
-                csi::run_grpc_servers(&mut [node_server, worker_server], async_server);
-            });
-            let async_fuse_thread = std::thread::spawn(move || {
-                if let Err(e) = async_fuse::start_async_fuse(etcd_delegate_clone, &args_clone) {
-                    panic!("failed to start async fuse, error is {:?}", e);
-                }
-            });
-            csi_thread
-                .join()
-                .unwrap_or_else(|e| panic!("csi thread error: {:?}", e));
-            async_fuse_thread
-                .join()
-                .unwrap_or_else(|e| panic!("csi thread error: {:?}", e));
-        }
-        RunAsRole::SchedulerExtender => {
-            let scheduler_extender = SchdulerExtender::new(
-                Arc::<MetaData>::clone(&md),
-                SocketAddr::new(args.ip_address, args.scheduler_extender_port),
-            );
-            let scheduler_extender_thread = std::thread::spawn(move || {
-                scheduler_extender.start();
-            });
-            scheduler_extender_thread
-                .join()
-                .unwrap_or_else(|e| panic!("scheduler extender error: {:?}", e));
-        }
-        RunAsRole::AsyncFuse => {
-            if let Err(e) = async_fuse::start_async_fuse(etcd_delegate_clone, &args_clone) {
+    let matches = parse_args();
+    if let Some(matches) = matches.subcommand_matches("start_csi_controller") {
```

Comment:
Define all subcommands as const string

Reply:
Due to clap, match subcommand must use original str.